src/charconv.c

```diff
@@ -72,6 +72,10 @@ static int wchar_string_to_cp850_string(char *out, const wchar_t *in, unsigned i
             return 0;
         }
     }
+    if (in[i]) {
+        fprintf(stderr, "Cannot convert input string '%ls' to 'CP850': String is too long\n", in);
```

Comment:
Maybe you should reuse the `CP%d` string here, and append hardcoded 850 as an argument (or actually `DEFAULT_DOS_CODEPAGE`).

Reply:
Ok, no problem, I can change it.

Reply:
Hm... looking at it and I think it does not make sense to replace it with `DEFAULT_DOS_CODEPAGE`. This function is named `wchar_string_to_cp850_string` (and not `to_default_string`) so it should print 850.

Reply:
Well, you already merged this, but it might make some sense to make up your mind on the extent to which you want to hardcode the choice of the code page. Either make it ‘internally generic’ (so that you only need to swap the code page table and number to change it) or remove the `DEFAULT_DOS_CODEPAGE` macro.

I’d suggest the former; it should make ditching iconv later (if you so choose) a bit easier.

Reply:
charconv contains fallback cp850 codepage which is ised when iconv is not available or when dosfstools was compiled without iconv support. This metioned function implements cp850 conversion, not generic _default_ conversion. So error message in function is correct.